### PR TITLE
Topic/multiple objects

### PR DESCRIPTION
**Pull request by gergondet**
Hi,

This pull request implements multiple objects recognition with a single instance of a recognition/tracking nodes pair. It tries to share as much computation between objects as possible in order to maintain good performance. However, on lower end systems, the detector node can slow up the tracking node significantly.

It is also possible (through service calls) to enable/disable tracking for certain objects. However, right now it is not possible to:
- train multiple objects at the same time (that wouldn't be very useful anyway I think)
- add objects on the fly and/or give an initial set of objects to track, all objects that you wish to track should be present in the configuration file at the beginning and they will all be tracked by default. Both of these options may be interesting to add later on.
- single-shot mode only works with one object

Best regards,

### CODE REVIEW

**Comment by po1** [comment]
Woah, that is BIG. But thanks for contributing!
@bmagyar good luck reviewing that. Let me know if I can help.

**Comment by awesomebytes** [comment]
Amazing!
Thank you @gergondet !

**Comment by bmagyar** [comment]
It's gonna be alright, I'm still familiar with the hidden goodies of BLORT this PR unlocks ;)

File: blort/include/blort/Recognizer3D/Recognizer3D.h [hunk 1]
```diff
@@ -64,18 +64,30 @@ class Recognizer3D{
...
+    /** @brief Creates a model for the training phase */
+    void initTrainingModel();
+
 	
 	/** @brief add sift features to sift model of an object
 	*   @param tFrame image/pixel map to search for new sift features
 	*   @param model shape description of model using faces and vertices (see TomGine::tgModel)
 	*   @param pose pose of the model as seen in the image */
-        bool learnSifts(IplImage* tFrame, const TomGine::tgModel& model, const TomGine::tgPose& pose);
+    bool learnSifts(IplImage* tFrame, const TomGine::tgModel& model, const TomGine::tgPose& pose);
```
Comment (bmagyar) [inline blort/include/blort/Recognizer3D/Recognizer3D.h]: could you fix the indentation in this file? it looks like some tabs got introduced here

File: blort_ros/bin/tracking.ini [hunk 2]
```diff
@@ -19,8 +19,8 @@ TexturePath = Resources/texture/
 ShaderPath = Tracker/shader/
 
 [Files]
-Model = Pringles.ply
```
Comment (bmagyar) [inline blort_ros/bin/tracking.ini:22]: I'd keep the Pringles as the default :)

**Comment by bmagyar** [comment]
Looks good to me, but I'll wait with this for #4.

A minor style question: separating the class in tracker_node.cpp made things much cleaner, the size of that one was getting a bit out of control. What do you think about keeping the contents of detector_node.cpp still in the cpp file? I'd prefer to not to have a new header as long as the contents in the .cpp are of manageable size.

**Comment by gergondet** [comment]
Hi,

I've fixed the two issues raised above.

Regarding the split, I think it's better to do it now as the node is still simple rather than if/when it gets more complex.

**Comment by bmagyar** [comment]
Okay, could you please first rebase this pull req on the new master? I merged #4.
There's a bit more indentation in the file https://github.com/gergondet/perception_blort/blob/61631c8bee2daf53054457b1ef4791f145f1f57d/blort/include/blort/Recognizer3D/Recognizer3D.h -- github shows the difference nicely

**Comment by gergondet** [comment]
Woops, apparently I had forgot to send a message here to indicate that I had rebased the PR on the latest master.

Best,

**Comment by bmagyar** [comment]
Thanks! I was also quite slow to check it, will try to test it today then merge.

**Comment by bmagyar** [comment]
No behavior change found, merging.

**Pull request merged**